src/account.ts

```diff
@@ -442,6 +446,49 @@ export abstract class Account {
             maxFee: BigInt(maxFee)
         });
     }
+
+    public async declareV2(
```

Comment:
I think the user should only have one declare method at their disposal. So this can be a private method which is called from the regular `declare` in case the contract factory provided is cairo1.

Reply:
https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/328/commits/da1e614d8cf66974c1affe826b3f0290c4c304b8